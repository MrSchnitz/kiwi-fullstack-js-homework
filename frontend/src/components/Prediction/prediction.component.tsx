import React from "react";
import {Button} from "@kiwicom/orbit-components/lib";
import {PredictionStyles} from "./prediction.styles";

export interface PredictionProps {
  text: string;
  onClick: () => void;
}

const Prediction: React.FC<PredictionProps> = ({
  text,
  onClick,
}: PredictionProps) => {
  return (
    <PredictionStyles>
      <Button
        type={"primarySubtle"}
        size={"small"}
        circled={true}
        onClick={onClick}
      >
        {text}
      </Button>
    </PredictionStyles>
  );
};

export default Prediction;
