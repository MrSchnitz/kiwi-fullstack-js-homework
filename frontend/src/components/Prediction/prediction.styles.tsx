import styled from "styled-components";

export const PredictionStyles = styled.div`
  margin: 0.5rem 0.25rem;
`;
