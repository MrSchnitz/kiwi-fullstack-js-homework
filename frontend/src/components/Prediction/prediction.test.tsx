import React from "react";
import { shallow, ShallowWrapper } from "enzyme";
import Prediction, { PredictionProps } from "./prediction.component";

describe("Prediction component", () => {
  let wrapper: ShallowWrapper;
  let mockOnClick: any;

  beforeEach(() => {
    mockOnClick = jest.fn();

    const mockProps: PredictionProps = {
      onClick: mockOnClick,
      text: "good",
    };

    wrapper = shallow(<Prediction {...mockProps} />);
  });

  it("should render Prediction component", () => {
    expect(wrapper).toMatchSnapshot();
  });

  it("should be called when component is clicked", () => {
    wrapper.find("Button").simulate("click");

    expect(mockOnClick).toHaveBeenCalled();
  });

  it("should display text", () => {
    expect(wrapper.find("Button").text()).toBe("good");
  });
});
