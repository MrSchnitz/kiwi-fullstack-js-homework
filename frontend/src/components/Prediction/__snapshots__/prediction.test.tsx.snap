// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Prediction component should render Prediction component 1`] = `ShallowWrapper {}`;
